docs/get-started/quickstarts/drive-rover.md

```diff
@@ -21,73 +23,126 @@ date: "2022-12-08"
 
-The easiest way to get started writing an application with Viam is to navigate to your [machine's page on the Viam app](https://app.viam.com/robots), select the **CONNECT** tab's **Code sample** page, then select **Python** and copy the boilerplate code.
+If you are using the Python SDK, [set up a virtual environment](/sdks/python/python-venv/) to package the SDK inside before running your code, avoiding conflicts with other projects or your system.
 
-{{% snippet "show-secret.md" %}}
+For macOS (both Intel `x86_64` and Apple Silicon) or Linux (`x86`, `aarch64`, `armv6l`), run the following commands:
+
+```sh {class="command-line" data-prompt="$"}
+python3 -m venv .venv
+source .venv/bin/activate
+pip install viam-sdk
+```
+
+Windows is not supported.
+If you are using Windows, use the [Windows Subsystem for Linux (WSL)](https://learn.microsoft.com/en-us/windows/wsl/install) and install the Python SDK using the preceding instructions for Linux.
+For other unsupported systems, see [Installing from source](https://python.viam.dev/#installing-from-source).
+
+If you intend to use the [ML (machine learning) model service](/services/ml/), use the following command instead, which installs additional required dependencies along with the Python SDK:
+
+```sh {class="command-line" data-prompt="$"}
+pip install 'viam-sdk[mlmodel]'
+```
+
+{{% /tab %}}
+{{% tab name="Go" %}}
+
+```sh {class="command-line" data-prompt="$"}
+go get go.viam.com/rdk/robot/client
+```
+
+{{% /tab %}}
+{{% tab name="TypeScript" %}}
+
+```sh {class="command-line" data-prompt="$"}
+npm install @viamrobotics/sdk
+```
+
+{{< alert title="Info" color="info" >}}
+The TypeScript SDK currently only supports building web browser apps.
+{{< /alert >}}
+
+{{% /tab %}}
+{{% tab name="C++" %}}
+
+Follow the [instructions on the GitHub repository](https://github.com/viamrobotics/viam-cpp-sdk/blob/main/BUILDING.md).
+
+{{% /tab %}}
+{{% tab name="Flutter" %}}
 
-This code snippet imports all the necessary libraries and sets up a connection with the Viam app in the cloud.
+```sh {class="command-line" data-prompt="$"}
+flutter pub add viam_sdk
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+{{< /expand >}}
+{{< expand "Step 3: Copy and run the sample code" >}}
+
+The sample code on the **CONNECT** tab, will show you how to authenticate and connect to a machine, as well as some of the methods you can use on your configured components and services.
 
-Next, create a file named <file>square.py</file> and paste the boilerplate code from the **Code sample** page of the Viam app into your file.
-Then, save your file.
+{{< tabs >}}
+{{% tab name="Python" %}}
 
-Run the code to verify that the Viam SDK is properly installed and that the `viam-server` instance on your robot is live.
+{{% snippet "show-secret.md" %}}
 
-You can run your code by typing the following into your terminal:
+Copy the code into a file called <FILE>square.py</FILE> and run the sample code to connect to your machine:
```

Comment:
This is longer, but I think "the code" makes me wonder "what code?"
Also consider moving the note down:
```suggestion
Go to the **CONNECT** tab and select **Python**.
Save your API key and API key ID as environment variables or include them in the code:

{{% snippet "show-secret.md" %}}

Copy and paste the code sample into a file called <FILE>square.py</FILE>.
Run the sample code to connect to your machine:
```
A shorter alternative:
```suggestion
Select **Python** and copy the sample code into a file called <FILE>square.py</FILE>.
Then run the code to connect to your machine:
```